pkg/printers/internalversion/printers_test.go

```diff
@@ -1268,14 +1268,12 @@ func TestPrintHumanReadableWithNamespace(t *testing.T) {
 	namespaceName := "testnamespace"
 	name := "test"
 	table := []struct {
-		obj          runtime.Object
-		isNamespaced bool
```

Comment:
I didn't expect these test changes as part of removing custom default printing

Reply:
For this test, the `isNamespaced == true` case calls `AddHandlers(printer)` on current line 1402. But for the `isNamespaced == false` case, `AddHandlers(printer)` was not called, which means a default printer handler is not set. This `isNamespaced == false` case contrives to force an error because there is no default printer handler, even though this is not the actual behavior of the printer. There is no case in which default printing will be missing. So I had to update these tests. For the objects that don't have namespaces (Namespace, Node), I suggest we remove those cases. Let me know what you think.

Reply:
I see. I think something like this would be a smaller change and would still exercise namespace/no-namespace printing:

```diff
diff --git a/pkg/printers/internalversion/printers_test.go b/pkg/printers/internalversion/printers_test.go
index c1b6b8c21f..0e4cee528a 100644
--- a/pkg/printers/internalversion/printers_test.go
+++ b/pkg/printers/internalversion/printers_test.go
@@ -1370,7 +1370,7 @@ func TestPrintHumanReadableWithNamespace(t *testing.T) {
                },
                {
                        obj: &api.PersistentVolume{
-                               ObjectMeta: metav1.ObjectMeta{Name: name, Namespace: namespaceName},
+                               ObjectMeta: metav1.ObjectMeta{Name: name},
                                Spec:       api.PersistentVolumeSpec{},
                        },
                        isNamespaced: false,
@@ -1417,30 +1417,23 @@ func TestPrintHumanReadableWithNamespace(t *testing.T) {
        }
 
        for i, test := range table {
+               printer := printers.NewTablePrinter(printers.PrintOptions{
+                       WithNamespace: true,
+                       NoHeaders:     true,
+               })
+               AddHandlers(printer)
+               buffer := &bytes.Buffer{}
+               err := printer.PrintObj(test.obj, buffer)
+               if err != nil {
+                       t.Fatalf("An error occurred printing object: %#v", err)
+               }
                if test.isNamespaced {
-                       // Expect output to include namespace when requested.
-                       printer := printers.NewTablePrinter(printers.PrintOptions{
-                               WithNamespace: true,
-                       })
-                       AddHandlers(printer)
-                       buffer := &bytes.Buffer{}
-                       err := printer.PrintObj(test.obj, buffer)
-                       if err != nil {
-                               t.Fatalf("An error occurred printing object: %#v", err)
-                       }
-                       matched := contains(strings.Fields(buffer.String()), fmt.Sprintf("%s", namespaceName))
-                       if !matched {
-                               t.Errorf("%d: Expect printing object to contain namespace: %#v", i, test.obj)
+                       if !strings.HasPrefix(buffer.String(), namespaceName+" ") {
+                               t.Errorf("%d: Expect printing object %T to contain namespace %q, got %s", i, test.obj, namespaceName, buffer.String())
                        }
                } else {
-                       // Expect error when trying to get all namespaces for un-namespaced object.
-                       printer := printers.NewTablePrinter(printers.PrintOptions{
-                               WithNamespace: true,
-                       })
-                       buffer := &bytes.Buffer{}
-                       err := printer.PrintObj(test.obj, buffer)
-                       if err == nil {
-                               t.Errorf("Expected error when printing un-namespaced type")
+                       if !strings.HasPrefix(buffer.String(), " ") {
+                               t.Errorf("%d: Expect printing object %T to not contain namespace got %s", i, test.obj, buffer.String())
                        }
                }
        }
```